src/node/rpc/frontend.h

```diff
@@ -552,15 +570,57 @@ namespace ccf
 
       auto rv = jsonrpc::pack(rep.value(), ctx.pack.value());
 
-#ifdef PBFT
-      if (history)
-        history->add_response(reqid, rv);
+      return rv;
 #endif
+    }
 
-      return rv;
+    /** Process a serialised command with the associated RPC context via PBFT
+     *
+     * If an RPC that requires writing to the kv store is processed on a
+     * follower, the serialised RPC is forwarded to the current network leader.
+     *
+     * @param ctx Context for this RPC
+     * @param input Serialised JSON RPC
+     */
+    std::vector<uint8_t> process_pbft(
+      enclave::RPCContext& ctx, const std::vector<uint8_t>& input) override
+    {
+      // TODO(#PBFT): Refactor this with process_forwarded().
+      Store::Tx tx;
+
+      auto pack = detect_pack(input);
+      if (!pack.has_value())
+        return jsonrpc::pack(
+          jsonrpc::error_response(
+            0, jsonrpc::ErrorCodes::INVALID_REQUEST, "Empty PBFT request."),
+          jsonrpc::Pack::Text);
+
+      auto rpc = unpack_json(input, pack.value());
+      if (!rpc.first)
+        return jsonrpc::pack(rpc.second, pack.value());
+
+      SignedReq signed_request;
+
+      // Strip signature
+      auto rpc_ = &rpc.second;
+      if (rpc_->find(jsonrpc::SIG) != rpc_->end())
+      {
+        auto& req = rpc_->at(jsonrpc::REQ);
+        rpc_ = &req;
+      }
+      auto& unsigned_rpc = *rpc_;
+
+      auto rep =
+        process_json(ctx, tx, ctx.fwd->caller_id, unsigned_rpc, signed_request);
+
+      // TODO(#PBFT): Add RPC response to history based on Request ID
```

Comment:
Why is that commented out?

Reply:
The `RequestID` is not yet passed down to `process_pbft()`. This should be a trivial change and we can address it when we start dealing with `history->add_response()`.